File: design-pattern-java/app/src/main/java/design/pattern/java/patterns/structural/composite/TextComponent.java
package design.pattern.java.patterns.structural.composite;

public class TextComponent implements UIComponent{
    private String message;
    public TextComponent(String message) {
        this.message = message;
    }
    @Override
    public String render() {
        return message;
    }
    public String getMessage() {
        return this.message;
    }
}

File: design-pattern-go/src/patterns/structural/composite/TextComponent.go
package composite

type TextComponent struct {
	message string
	UIComponent
}

func NewTextComponent(message string) TextComponent {
	return TextComponent{message: message}
}

func (c TextComponent) Render() string {
	return c.message
}
func (c *TextComponent) GetMessage() string {
	return c.message
}
